soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -168,6 +161,24 @@ bool z_arm_on_enter_cpu_idle(void)
 
 	return true;
 }
+
+bool z_arm_on_enter_cpu_idle(void)
+{
+	bool ok_to_sleep = nrf53_anomaly_160_check();
+
+#if (LOG_LEVEL >= LOG_LEVEL_DBG)
```

Comment:
I don't think that you need this `#ifdef`. Tried to compile without and ROM/RAM is the same.

Reply:
Yes. It looks like now, after moving the essential part of `z_arm_on_enter_cpu_idle()` to `nrf53_anomaly_160_check()`, gcc is able to optimize out the `suppress_message` variable when the message is not to be printed at all. It could not do it earlier, when the message handling was merged into the anomaly workaround code, so I added several such `#if`s. Then I reorganized the code a little to make it easier to read and I didn't check again the optimization.
But since this `#if` is not actually harmful and the code is quite clear, I'd rather keep it as it is now, just to not rely on optimization capabilities of compilers. Or do you foresee some issues that this `#if` may bring in the future?

Reply:
approved. I don't like ifdefs so if they can be avoided i'm all for it but i won't insist.